interfaces/builtin/basedeclaration.go

```diff
@@ -463,6 +463,12 @@ slots:
         - app
     deny-connection: true
     deny-auto-connection: true
+  uhid:
+    allow-installation:
+      slot-snap-type:
+        - core
+        - gadget
```

Comment:
This should only be for 'core', not 'gadget'. Because it is part of core, you need to add this to snap/implicit.go as well, in implicitSlots.